InnerEye/ML/deep_learning_config.py

```diff
@@ -256,35 +401,82 @@ class DeepLearningConfig(GenericConfig, CudaAwareConfig):
+
+
+class TrainerParams(CudaAwareConfig):
+    num_epochs: int = param.Integer(100, bounds=(1, None), doc="Number of epochs to train.")
     recovery_checkpoint_save_interval: int = param.Integer(10, bounds=(0, None),
                                                            doc="Save epoch checkpoints when epoch number is a multiple "
                                                                "of recovery_checkpoint_save_interval. The intended use "
                                                                "is to allow restore training from failed runs.")
-    train_batch_size: int = param.Integer(4, bounds=(0, None),
-                                          doc="The number of crops that make up one minibatch during training.")
     detect_anomaly: bool = param.Boolean(False, doc="If true, test gradients for anomalies (NaN or Inf) during "
                                                     "training.")
     use_mixed_precision: bool = param.Boolean(False, doc="If true, mixed precision training is activated during "
                                                          "training.")
+    max_num_gpus: int = param.Integer(default=-1, doc="The maximum number of GPUS to use. If set to a value < 0, use"
+                                                      "all available GPUs.")
+    pl_num_sanity_val_steps: int = \
+        param.Integer(default=0, doc="PyTorch Lightning trainer flag 'num_sanity_val_steps': Number of validation "
+                                     "steps to run before training, to identify possible problems")
+    pl_deterministic: bool = \
+        param.Integer(default=True,
+                      doc="Controls the PyTorch Lightning trainer flags 'deterministic' and 'benchmark'. If "
+                          "'pl_deterministic' is True, results are perfectly reproducible. If False, they are not, but "
+                          "you may see training speed increases.")
+    start_epoch: int = param.Integer(0, bounds=(0, None), doc="The first epoch to train. Set to 0 to start a new "
+                                                              "training. Set to a value larger than zero for starting"
+                                                              " from a checkpoint.")
+
+
+class DeepLearningConfig(WorkflowParams,
+                         DatasetParams,
+                         OutputParams,
+                         OptimizerParams,
+                         TrainerParams,
+                         CudaAwareConfig,
+                         GenericConfig):
+    """
+    A class that holds all settings that are shared across segmentation models and regression/classification models.
+    """
+    _model_category: ModelCategory = param.ClassSelector(class_=ModelCategory,
+                                                         doc="The high-level model category described by this config.")
+
+    num_dataload_workers: int = param.Integer(8, bounds=(0, None),
```

Comment:
For num_dataload_workers, train_batch_size, shuffle, pin_memory we could have another Param class classed "DataLoaderParams".

Reply:
I only moved out params in separate classes if they are also used in the container models. num_dataload_workers is something that is only used in the InnerEye models, and hence I felt it could stay here.